### ISSUE TITLE
snmp_facts module incorrect report ifSpeed

### ISSUE THREAD

**Issue by odebree** [ansible/ansible#43421]
##### SUMMARY

snmp_facts module incorrectly reports ifSpeed (SNMP OID 1.3.6.1.2.1.2.2.1.5) with MTU value 

bug is present in snmp_facts.py line 328:

if v.ifMtu in current_oid: should be replaced with if v.ifSpeed in current_oid:

##### ISSUE TYPE
 - Bug Report

##### ANSIBLE VERSION

```
since 1.9 - 2.6.2
```

##### OS / ENVIRONMENT

all environments 
##### STEPS TO REPRODUCE

```yaml
---

- hosts: IOS-HOST1
  connection: local
  gather_facts: no
  strategy: debug
  vars:
    ansible_vrfs:
      -



  pre_tasks:
    - import_tasks: pb_auth.yml
      tags:
        - build
        - deploy



    - name: "SNMP FACTS"
      snmp_facts:
       host: '{{ ansible_ssh_host }}'
       version: v2c
       community: public
      delegate_to: localhost
```

##### EXPECTED RESULTS

           "4": {
                "adminstatus": "up",
                "description": "",
                "ifindex": "4",
                "mac": "08002770615a",
                "mtu": "1514",
                "name": "GigabitEthernet0/0/0/2",
                "operstatus": "up",
                **"speed": "1000000000"**
            },
##### ACTUAL RESULTS

            "4": {
                "adminstatus": "up",
                "description": "",
                "ifindex": "4",
                "mac": "08002770615a",
                "mtu": "1514",
                "name": "GigabitEthernet0/0/0/2",
                "operstatus": "up",
                **"speed": "1514"**
            },

```

```

**Comment by odebree** [comment]
this looks to be a long standing bug as I can see the typo in the pre 1.9 repo on githup

**Comment by odebree** [comment]
Please see patched file, along with enhancement to include snmp_ifType (OID 1.3.6.1.2.1.2.2.1.3)

tested against ansible 2.6.2
[snmp_facts.py.zip](https://github.com/ansible/ansible/files/2241086/snmp_facts.py.zip)

**Comment by ericwb** [comment]
This was fixed in pull: https://github.com/ansible/ansible/pull/39578

**Comment by mkrizek** [comment]
> This was fixed in pull: #39578

Closing as per above.
